engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/EscalationEventValidationTest.java

```diff
@@ -0,0 +1,469 @@
+    assertThat(deployedProcess.getProcessDefinitionKey()).isGreaterThan(0);
+  }
+
+  @Test
+  public void shouldDeployProcessModelWithEscalationEndEvent() {
+    // given
+    final BpmnModelInstance processDefinition =
+        Bpmn.createExecutableProcess()
+            .startEvent()
+            .endEvent("end", builder -> builder.escalation("escalationCode"))
+            .done();
+
+    // when
+    final Record<DeploymentRecordValue> result =
+        ENGINE.deployment().withXmlResource(processDefinition).deploy();
+
+    // then
+    assertThat(result.getKey()).isGreaterThan(0);
+    assertThat(result.getValue().getProcessesMetadata()).hasSize(1);
+    final ProcessMetadataValue deployedProcess = result.getValue().getProcessesMetadata().get(0);
+    assertThat(deployedProcess.getVersion()).isEqualTo(1);
+    assertThat(deployedProcess.getProcessDefinitionKey()).isGreaterThan(0);
+  }
+
+  @Test
+  public void shouldDeployProcessModelWithEscalationBoundaryEvent() {
+    // given
+    final BpmnModelInstance processDefinition =
+        Bpmn.createExecutableProcess()
+            .startEvent()
+            .callActivity("call", builder -> builder.zeebeProcessId("sub"))
+            .boundaryEvent("catch", builder -> builder.escalation("escalationCode"))
+            .endEvent()
+            .done();
+
+    // when
+    final Record<DeploymentRecordValue> result =
+        ENGINE.deployment().withXmlResource(processDefinition).deploy();
+
+    // then
+    assertThat(result.getKey()).isGreaterThan(0);
+    assertThat(result.getValue().getProcessesMetadata()).hasSize(1);
+    final ProcessMetadataValue deployedProcess = result.getValue().getProcessesMetadata().get(0);
+    assertThat(deployedProcess.getVersion()).isEqualTo(1);
+    assertThat(deployedProcess.getProcessDefinitionKey()).isGreaterThan(0);
+  }
+
+  @Test
+  public void
+      shouldRejectDeploymentIfEscalationBoundaryEventIsNotAttachedToChildProcessOrCallActivity() {
+    // given
+    final BpmnModelInstance processDefinition =
+        Bpmn.createExecutableProcess()
+            .startEvent()
+            .serviceTask("task", builder -> builder.zeebeJobType("type"))
+            .boundaryEvent("catch", AbstractBoundaryEventBuilder::escalation)
+            .endEvent()
+            .done();
+    // when
+    final Record<DeploymentRecordValue> rejectedDeployment =
+        ENGINE.deployment().withXmlResource(processDefinition).expectRejection().deploy();
+
+    // then
+    Assertions.assertThat(rejectedDeployment)
+        .hasRecordType(RecordType.COMMAND_REJECTION)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT)
+        .hasRejectionReason(
+            "Expected to deploy new resources, but encountered the following errors:\n"
+                + "'process.xml': - Element: catch\n"
+                + "    - ERROR: "
+                + "An escalation boundary event should only be attached to a subprocess, or a call activity.\n");
```

Comment:
🔧 Sorry I didn't realise last review that this rejection reason came with so much noise 😅 

It's probably more readable if we change it to:
```java
    Assertions.assertThat(rejectedDeployment)
        .hasRecordType(RecordType.COMMAND_REJECTION)
        .hasRejectionType(RejectionType.INVALID_ARGUMENT);
    assertThat(rejectedDeployment.getRejectionReason())
        .contains(
            "An escalation boundary event should only be attached to a subprocess, or a call activity.");
```